### PR TITLE
feat: Add the notification templates module to node package

### PR DESCRIPTION
**Pull request by codingmickey**
- **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
Feature of integrating the notification templates API with the node module for easier access
- **Why was this change needed?** (You can also link to an open issue here)
Completes #1320 
- **Other information**:

### CODE REVIEW

File: packages/node/src/lib/notification-template/notification-template.interface.ts [hunk 1]
```diff
@@ -0,0 +1,32 @@
+import {
+  INotificationTemplateStep,
+  IPreferenceChannels,
+} from 'libs/shared/dist';
+
+import { AxiosPromise } from '.pnpm/axios@0.26.1/node_modules/axios';
+
+export interface INotificationTemplates {
```
Comment (florian-lefebvre) [inline packages/node/src/lib/notification-template/notification-template.interface.ts:8]: Don't specify the AxiosPromise as a result. So you'll also have to remove the import above

File: packages/node/src/lib/notification-template/notification-template.ts [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+export class NotificationTemplates
+  extends WithHttp
+  implements INotificationTemplates
+{
+  /**
+   * @param {number} page - Page number to fetch
+   * @param {number} limit - Number of results to fetch in one page
+   * @returns {Promise<AxiosResponse>} - Fetches all the notification-templates with the given parameters
+   */
+  async getAll(page?: number, limit?: number): Promise<AxiosResponse> {
```
Comment (florian-lefebvre) [inline packages/node/src/lib/notification-template/notification-template.ts:19]:
```suggestion
  async getAll(page?: number, limit?: number) {
```

File: packages/node/src/lib/notification-template/notification-template.ts [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+   * @param {string} name - Name of the notification-template to create
+   * @param {string} notificationGroupId - Id of the group that it belongs to
+   * @param {INotificationTemplatePayload} data - All the additional parameters to create a notification-template
+   * @returns {Promise<AxiosResponse>} - Creates the notification-template with the given parameters
+   */
+  async create(
+    name: string,
+    notificationGroupId: string,
+    data: INotificationTemplatePayload
+  ): Promise<AxiosResponse> {
```
Comment (florian-lefebvre) [inline packages/node/src/lib/notification-template/notification-template.ts:38]:
```suggestion
  ) {
```

File: packages/node/src/lib/notification-template/notification-template.ts [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+   * @param {string} templateId - templateId of the notification-template to update
+   * @param {string} name - Name of the notification-template to update
+   * @param {INotificationTemplatePayload} data - All the additional parameters to create a notification-template
+   * @returns {Promise<AxiosResponse>} - Creates the notification-template with the given parameters
+   */
+  async update(
+    templateId: string,
+    name: string,
+    data: INotificationTemplatePayload
+  ): Promise<AxiosResponse> {
```
Comment (florian-lefebvre) [inline packages/node/src/lib/notification-template/notification-template.ts:56]:
```suggestion
  ) {
```

File: packages/node/src/lib/notification-template/notification-template.ts [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+      name,
+      ...data,
+    });
+  }
+
+  /**
+   * @param {string} templateId - templateId of the notification-template to delete
+   * @returns {Promise<AxiosResponse>} - Creates the notification-template with the given parameters
+   */
+  async delete(templateId: string): Promise<AxiosResponse> {
```
Comment (florian-lefebvre) [inline packages/node/src/lib/notification-template/notification-template.ts:67]:
```suggestion
  async delete(templateId: string) {
```

File: packages/node/src/lib/notification-template/notification-template.ts [hunk 6]
```diff
@@ -0,0 +1,90 @@
...
+    return await this.http.delete(`notification-templates/${templateId}`);
+  }
+
+  /**
+   * @param {string} templateId - templateId of the notification-template to get details of
+   * @returns {Promise<AxiosResponse>} - Creates the notification-template with the given parameters
+   */
+  async getOne(templateId: string): Promise<AxiosResponse> {
```
Comment (florian-lefebvre) [inline packages/node/src/lib/notification-template/notification-template.ts:75]:
```suggestion
  async getOne(templateId: string) {
```

File: packages/node/src/lib/notification-template/notification-template.ts [hunk 7]
```diff
@@ -0,0 +1,90 @@
+import { AxiosInstance, AxiosResponse } from 'axios';
```
Comment (florian-lefebvre) [inline packages/node/src/lib/notification-template/notification-template.ts:1]:
```suggestion
import axios from 'axios';
```
Won't be used in the code but required by `WithHttp` to get types

File: packages/node/src/lib/notification-template/notification-template.ts [hunk 8]
```diff
@@ -0,0 +1,90 @@
...
+} from './notification-template.interface';
+
+export class NotificationTemplates
+  extends WithHttp
+  implements INotificationTemplates
+{
+  /**
+   * @param {number} page - Page number to fetch
+   * @param {number} limit - Number of results to fetch in one page
+   * @returns {Promise<AxiosResponse>} - Fetches all the notification-templates with the given parameters
```
Comment (florian-lefebvre) [inline packages/node/src/lib/notification-template/notification-template.ts:17]: I won't comment for every method but please remove this line

File: packages/node/src/lib/notification-template/notification-template.interface.ts [hunk 9]
```diff
@@ -0,0 +1,32 @@
...
+    templateId: string,
+    name: string,
+    data: INotificationTemplatePayload
+  ): AxiosPromise;
+  delete(templateId: string): AxiosPromise;
+  getOne(templateId: string): AxiosPromise;
+  updateStatus(templateId: string, active: boolean): AxiosPromise;
+}
+
+export interface INotificationTemplatePayload {
```
Comment (florian-lefebvre) [inline packages/node/src/lib/notification-template/notification-template.interface.ts:24]: Isn't anything missing? Like name, notificationGroupId...

**Review by florian-lefebvre — CHANGES_REQUESTED** [review]
Great work! I left some reviews but nothing crazy. However, there are no tests yet. You may be waiting to have the current code validated before adding them to the PR?

File: packages/node/src/lib/notification-template/notification-template.interface.ts (hunk 9 above)
Comment (codingmickey) [reply packages/node/src/lib/notification-template/notification-template.interface.ts:24]: That is included in whichever methods required, so I thought not to include in this interface

**Comment by codingmickey** [comment]
> ...You may be waiting to have the current code validated before adding them to the PR?

Yes

File: packages/node/src/lib/notification-template/notification-template.interface.ts (hunk 9 above)
Comment (florian-lefebvre) [reply packages/node/src/lib/notification-template/notification-template.interface.ts:24]: @scopsy wdyt?

**Comment by codingmickey** [comment]
Is anything left on my part, with respect to this issue? @florian-lefebvre

**Review by florian-lefebvre — APPROVED** [review]
LGTM

**Comment by florian-lefebvre** [comment]
I think that's okay, we just need advice on the unresolved discussion

**Comment by codingmickey** [comment]
> I think that's okay, we just need advice on the unresolved discussion

Yeah sure!

File: packages/node/src/lib/notification-template/notification-template.interface.ts (hunk 9 above)
Comment (scopsy) [reply packages/node/src/lib/notification-template/notification-template.interface.ts:18]: Yes looks like it's missing here. Sorry for the delayed response, my GitHub notification are going crazy.

@codingmickey would be best to specify the properties from here: https://docs.novu.co/api/create-notification-template/

Comment (codingmickey) [reply packages/node/src/lib/notification-template/notification-template.interface.ts:18]: Yes! I got the properties from there.

Oh okay, are you suggesting that I put all the parameters in a single template and then tell the user to put those parameters as the object with `name` and `notificationGroupId`(and not as a different parameter as I'm currently doing it?) as required?

Comment (scopsy) [reply packages/node/src/lib/notification-template/notification-template.interface.ts:14]: Looks great like this!

**Review by scopsy — APPROVED** [review]

**Pull request merged**